debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/ConnectionContext.java

```diff
@@ -272,6 +274,9 @@ public void execute(String desc, Consumer<MongoClient> operation) {
                     return;
                 } catch (Throwable t) {
                     errorHandler.accept(desc, t);
+                    if (!isRunning()) {
```

Comment:
If `stop` variable has `false` value (it is default value for `AtomicBoolean`), then `ConnectException` is thrown here, and `errorMetronome.pause()` will never be called.